### ISSUE TITLE
publishToConfluence: Does not update existing page

### ISSUE THREAD

**Issue by faltfe** [docToolchain/docToolchain#1206]
**Describe the bug**
When running `./dtcw.bat publishToConfluence -DconfluenceApiKey=Tw...3 --info` twice an error is thrown

```plain
something went wrong - got an http response code 400:
A page with this title already exists: A page already exists with the title Development Documentation in the space with key TC
please check the ancestorId in your config file

> Task :publishToConfluence FAILED
:publishToConfluence (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 3.818 secs.

FAILURE: Build failed with an exception.

* Where:
Script '/home/dtcuser/docToolchain/scripts/publishToConfluence.gradle' line: 30

* What went wrong:
Execution failed for task ':publishToConfluence'.
> java.lang.Exception: Parent does not exist
```

**To Reproduce**
1. Run `./dtcw.bat publishToConfluence -DconfluenceApiKey=Tw...3 --info` once
2. A new page is created in Confluence as expected underneath the given ancesterId
3. Run `./dtcw.bat publishToConfluence -DconfluenceApiKey=Tw...3 --info` the second time

**Expected behavior**
When running the task publishToConfluence again the previous created pages are updated when there are changes or the page is deleted and uploaded again. If there are no changes nothing should happen.

**Configuration**
 - docToolchain version: $version = "2.2.1", $dockerVersion = "2.2.1"
 - OS: Windows 11
 - Relevant SDK Versions OpenJDK 17
 - Atlassian Confluence 7.19.3 (confluence-3: 6ff93426)

```groovy
confluence.with {
    input = [
            [ file: "build/html5/development/index.html", ancestorName: '300 - Development', ancestorId: '334425388' ],
            [ file: "build/html5/arc42/index.html", ancestorId: "87511988", ancestorName: 'Documentation' ],
    ]
```

The `development/index.html` will look like

```html
:imagesdir: ../images/development
// ====================================
= Development Documentation
:description: Development documentation of the application.
:sectanchors:

//
// toc-title definition MUST follow document title without blank line!
:toc-title: Table of Contents
// enable table-of-contents
:toc:
:numbered:
:keywords: generated

include::getting-started/index.adoc[]
```

The Id of the parent page does exists, as the page was created successfully in the first run. The page information of Confluence has the url: https://url-to-confluence/confluence/pages/viewinfo.action?pageId=334425388

**Comment by arutscheidt** [comment]
~Same here~
EDIT: Had another problem

**Comment by faltfe** [comment]
The problem still happens with the latest version

dtcw 0.52 - ##DTCW_GIT_HASH##
docToolchain 3.0.2
OS/arch: pwsh windows x64
Available docToolchain environments: local docker
Environments with docToolchain [3.0.2]: docker
Using environment: docker

**Comment by faltfe** [comment]
With the latest version 3.1.1 this issue seems to be fixed. But now images cannot be published, see #1291

**Comment by PacoVK** [comment]
@faltfe  with the release of 3.1.2 both #1291  and this #1206 should be fixed now :) Please let me know if the issue persists

**Comment by faltfe** [comment]
@PacoVK with version 3.1.2 I can upload and replace existing pages. Images are also uploaded (#1291) with one major exception.

```plantuml
[plantuml,03_technical-context,svg]
....
@startuml
skinparam componentStyle uml2

foo <-u-> bar

@enduml
....
```

This image will **not** be uploaded. The error message says:

```
3. System Scope and Context
Could not find embedded image at a known location
Embedded Image Hash d219977e9106016aa4370ca771cd5267
Creating image at /project/build/html5/arc42//confluence/images/
    image: /project/build/html5/arc42/confluence/images/d219977e9106016aa4370ca771cd5267.svg xml
Could not find embedded image at a known location
Embedded Image Hash f67e09860e2ac609af3539f4780e64ca
Creating image at /project/build/html5/arc42//confluence/images/
    image: /project/build/html5/arc42/confluence/images/f67e09860e2ac609af3539f4780e64ca.svg xml
allPages already retrieved
pages retrieved
> created page 2316800867
```

As soon as I change `[plantuml,03_technical-context,png]` the images are uploaded. The console log prints

```
3. System Scope and Context
Could not find embedded image at a known location
Embedded Image Hash 7dc185efb3f92bfc5fec13af8a7d0b2f
    image: /project/build/html5/arc42/confluence/images/7dc185efb3f92bfc5fec13af8a7d0b2f.png
Could not find embedded image at a known location
Embedded Image Hash 8bfe223fc19e37321a4a61042169f9a4
Creating image at /project/build/html5/arc42//confluence/images/
    image: /project/build/html5/arc42/confluence/images/8bfe223fc19e37321a4a61042169f9a4.png
allPages already retrieved
pages retrieved
retrieving page with id 2316800867
found existing page: 2316800867 version 1
> updated page 2316800867
Uploading attachment to /confluence/rest/api/content/2316800867/child/attachment
Uploading attachment to /confluence/rest/api/content/2316800867/child/attachment
```

This seems to be another issue not directly related with the original issue.

**Comment by la-ko** [comment]
@faltfe, @PacoVK, I face the same issue as @faltfe when rendering PlantUML images as SVG. I analysed the problem locally which led me to https://github.com/docToolchain/docToolchain/blob/ng/core/src/main/groovy/org/docToolchain/scripts/asciidoc2confluence.groovy#L554 converting the plus sign (+) used in the file extension for MIME type `image/svg+xml` to a space.

This leads to the `java.io.FileNotFoundException`, later on. As I haven't completely delved into the code base and the design, I don't feel comfortable enough to hand in a PR, yet. Hope, a more experienced docTooolchain developer can help :).

**Comment by PacoVK** [comment]
Thank both of you for the investigation :) While this is a problem caused by embedding svg images, you should be able to workaround for now setting `failOnMissingImages=false` in you docToolchian config. This causes the converter not to inline all images in the HTML but reference the actual image. Meanwhile i check how/ if we can support inlined svg images 👍

**Comment by la-ko** [comment]
@PacoVK thanks for the hint. This did the trick for me 😄.

**Comment by PacoVK** [comment]
while the initial bug was solved, i created another issue for the svg bug (and already fixed it). Closing this one as resolved, please watch #1300  for embedded svg support
